src/main/java/io/kestra/plugin/aws/kinesis/PutRecords.java

```diff
@@ -0,0 +1,295 @@
+                builder.streamARN(streamArn);
+            } else if (!Strings.isNullOrEmpty(streamName)) {
+                builder.streamName(streamName);
+            } else {
+                throw new IllegalArgumentException("Either streamName or streamArn has to be set.");
+            }
+
+
+            List<PutRecordsRequestEntry> putRecordsRequestEntryList = records.stream()
+                .map(throwFunction(record -> record.toPutRecordsRequestEntry(runContext)))
+                .collect(Collectors.toList());
+
+
+            builder.records(putRecordsRequestEntryList);
+            return client.putRecords(builder.build());
+        }
+    }
+
+    private List<Record> getRecordList(Object records, RunContext runContext) throws IllegalVariableEvaluationException, URISyntaxException, IOException {
+        if (records instanceof String) {
+            // Try to deserialize json string
+            try {
+                return MAPPER.readValue((String) records, new TypeReference<List<Record>>() {
+                });
+            } catch (Exception e) {
+                // If not json try to read from it as an URI
+            }
+
+            URI from = new URI(runContext.render((String) records));
+            if (!from.getScheme().equals("kestra")) {
+                throw new IllegalArgumentException("Invalid records parameter, must be a Kestra internal storage URI, or a list of record.");
+            }
+            try (BufferedReader inputStream = new BufferedReader(new InputStreamReader(runContext.uriToInputStream(from)))) {
+                return Flowable.create(FileSerde.reader(inputStream, Record.class), BackpressureStrategy.BUFFER)
+                    .toList().blockingGet();
+            }
+        } else if (records instanceof List) {
+            return MAPPER.convertValue(records, new TypeReference<>() {
+            });
+        }
+
+        throw new IllegalVariableEvaluationException("Invalid records type '" + records.getClass() + "'");
+    }
+
+    private File writeOutputFile(RunContext runContext, PutRecordsResponse putRecordsResponse, List<Record> records) throws IOException {
+        // Create Output
+        File tempFile = runContext.tempFile(".ion").toFile();
+        try (var stream = new FileOutputStream(tempFile)) {
+            List<PutRecordsResultEntry> responseEntries = putRecordsResponse.records();
+            for (int i = 0; i < responseEntries.size(); i++) {
+                PutRecordsResultEntry responseEntry = responseEntries.get(i);
+                PutRecords.OutputEntry entry = OutputEntry.builder()
+                    .record(records.get(i))
+                    .sequenceNumber(responseEntry.sequenceNumber())
+                    .shardId(responseEntry.shardId())
+                    .errorCode(responseEntry.errorCode())
+                    .errorMessage(responseEntry.errorMessage())
+                    .build();
+                FileSerde.write(stream, entry);
```

Comment:
Its not mandatory, but its way better if you follow the convention which is using flowable.
Here is an example : https://github.com/kestra-io/plugin-serdes/blob/4411bd47e8815964c6841d47a06809feb0fda070/src/main/java/io/kestra/plugin/serdes/csv/CsvReader.java#L101-L101

Reply:
yeah I wanted to use it initially but I need the index to be able to retrieve the corresponding record from the original list. Is there a way to do that ?

Reply:
If you have a working example I'll take it I could not find a way to make it... I'm not as fluent as I'd want with flowable :sweat:

Reply:
Don't worry, keeping the index with flowable is complex

Reply:
In the end I managed to make something working with a zipWith ;)